services/node_stake_update.proto

```diff
@@ -37,34 +37,43 @@ message NodeStakeUpdateTransactionBody {
    */
   Timestamp end_of_staking_period = 1;
 
-  /**
-   * The daily reward rate for this staking period in tinybars
-   */
-  int64 reward_rate = 2;
-
   /**
    * Staking info of each node at the beginning of the new staking period
    */
-  repeated NodeStake node_stake = 3;
+  repeated NodeStake node_stake = 2;
 }
 
 /**
  * Staking info for each node for the staking period that is ending
  */
 message NodeStake {
-
   /**
    * Node id of this node
    */
   int64 node_id = 1;
 
   /**
-   * Consensus weight for this node, at the end of current staking period.
+   * Consensus weight of this node for the new staking period.
    */
   int64 stake = 2;
 
   /**
-   * Total of balance of all accounts staked to this node that have declineReward set to false, at the end of staking period.
+   * Total of (balance + stakedToMe) for all accounts staked to this node with declineReward=false, at the beginning of the new staking period.
    */
   int64 stake_rewarded = 3;
+
+  /**
+   * The reward rate _per whole hbar_ that was staked to this node with declineReward=false from the start of 
+   * the staking period that is ending. 
+   * 
+   * There are three main cases to consider:
+   *   1. The reward rate is zero because the node's starting stake was below its minStake.
+   *   2. The node's stake was in the range [minStake, maxStake], and this is the "base" reward rate; namely,
+   *           baseRewardRate = totalRewardRate / (totalStakeRewardStart / 100_000_000)
+   *   3. The node's stake exceeded its maxStake, so this is a down-scaling of the base rate:
+   *           scaledRewardRate = (baseRewardRate * stakeToReward) / maxStake
```

Comment:
I think the condition for 3 should be: the node's stakeToReward at the beginning of the ending staking period (let's name it `stakeToRewardStartNotCapped`) exceeds its maxStake. And to downscale it, we should do

```
scaledRewardRate = baseRewardRate * maxStake / stakeToRewardStartNotCapped 
```

Reply:
But perhaps we don't need to list the 3 conditions as these are internal implementation details? We would only need to list details that would cause consumers to have to react differently, but here I think for all 3 the mirror node would still simply store the value and aggregate for use in a pending reward calculation.

Reply:
Haha @xin-hedera I was trying to just mindlessly copy from [the source](https://github.com/hashgraph/hedera-services/blob/redirect-deleted-account-rewards/hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleStakingInfo.java#L212) and somehow inverted the fraction in the process. 😝 

Deleted these implementations details and rearranged fields alphabetically.